File: example/src/main/java/com/example/circularreveal/CustomActivity2.java
package com.example.circularreveal;

import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.graphics.Rect;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.hendraanggrian.widget.CircularRevealFrameLayout;

import butterknife.BindView;
import butterknife.ButterKnife;

public class CustomActivity2 extends AppCompatActivity {

    public final static String EXTRA_RECT = "com.example.circularreveal.CustomActivity2";

    @BindView(R.id.circularrevealframelayout) CircularRevealFrameLayout layout;
    @BindView(R.id.textview) TextView textView;
    private Rect rect;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_custom2);
        ButterKnife.bind(this);
        rect = getIntent().getParcelableExtra(EXTRA_RECT);
        textView.post(new Runnable() {
            @Override
            public void run() {
                layout.create(textView, rect.centerX(), rect.centerY()).start();
            }
        });
    }

    @Override
    public void onBackPressed() {
        Animator animator = layout.create(textView, rect.centerX(), rect.centerY(), true);
        animator.addListener(new AnimatorListenerAdapter() {
            @Override
            public void onAnimationEnd(Animator animation) {
                textView.setVisibility(View.INVISIBLE);
                finish();
                overridePendingTransition(0, 0);
            }
        });
        animator.start();
    }
}

File: example/src/main/java/com/example/reveallayout/CustomActivity2.kt
package com.example.reveallayout

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.graphics.Rect
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.hendraanggrian.reveallayout.Radius
import kotlinx.android.synthetic.main.activity_custom2.*

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
class CustomActivity2 : AppCompatActivity() {

    companion object {
        val EXTRA_RECT = "com.example.circularreveal.CustomActivity2"
    }

    private var rect: Rect? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_custom2)
        rect = intent.getParcelableExtra<Rect>(EXTRA_RECT)
        textView.post {
            layout.reveal(textView, rect!!.centerX(), rect!!.centerY(), Radius.NONE_LARGE)
                    .start()
        }
    }

    override fun onBackPressed() {
        layout.reveal(textView, rect!!.centerX(), rect!!.centerY(), Radius.LARGE_NONE).apply {
            addListener(object : AnimatorListenerAdapter() {
                override fun onAnimationEnd(animation: Animator) {
                    textView.visibility = View.INVISIBLE
                    finish()
                    overridePendingTransition(0, 0)
                }
            })
            start()
        }
    }
}